libmscore/note.cpp

```diff
@@ -2358,18 +2358,36 @@ QString Note::noteTypeUserName() const
 //---------------------------------------------------------
 
 void Note::scanElements(void* data, void (* func)(void*, Element*), bool all)
+{
+    if (all || visible() || score()->showInvisible()) {
+        func(data, this);
+    }
+    if (treeChildCount() > 0) {
+        Element::scanElements(data, func, all);
+    }
+    // TODO remove this:
+    if (tieBack()) {
+        tieBack()->scanElements(data, func, all);
```

Comment:
I'm just checking this one, it may not be necessary. (see https://github.com/musescore/MuseScore/pull/6318/files#diff-32e6dbba79e3bb125b29f1ec59ab3719L2264-R2268)

Reply:
I expect it will be necessary because you might have a tie that goes over a page, and you want to do layout for only the 2nd page. You might need to run MuseScore to check this case rather than rely on a test.

Reply:
Well if the tie segments are children of the system then I suppose this might not be needed after all. Perhaps it's not even necessary to scan the forward tie...

Reply:
> I expect it will be necessary because you might have a tie that goes over a page, and you want to do layout for only the 2nd page. You might need to run MuseScore to check this case rather than rely on a test.
I checked it, you were right. It doesn't show ties on the 2nd page without this line.

Reply:
That suggests something else is wrong then.  That tie segment on the second page belongs to a system as well, why isn’t it laid out as part of that?

Reply:
For the record, the line is already gone for 3.5, and this problem doesn’t occur there with or without the line.  But other problems do occur with it, because it does cause the tie to get scanned twice.

Reply:
> That suggests something else is wrong then.

Yes, and I just found out what else is wrong, haha..

In the current tree model, I have made spanner segments children of the spanner. That means that the tie contains the tie segments. ScoreView is using Page::scanElements to draw each page, so a tie on a note on Page 1 is not getting drawn on page 2. Scanning `tieBack` along with `tieFor` makes the tie get drawn on Page 2. But then I looked at the other spanners: slurs, for example, don't get drawn on Page 2, and get drawn on the same page instead. Even with this line there is a bug, the tie gets drawn on both page 1 and page 2.

So maybe I should scan the spanner segments in systems in the old way (by overriding `System::scanElements`)? Or should I change the tree model to make the system contain the spanner segments?

Reply:
I think a ton of things will break if you change the basic assumption that spanner segments are children of the system, it's fundamental to the whole reason spanners are broken up into segments to begin with.  Maybe there's a way, with a lots more code, to re-implement the spanner functionality to change this, but I would strongly advise against trying.

Reply:
Interesting. I didn't know that spanner segments were owned by the system, but it does make sense.

@Kartikay26 
> So maybe I should scan the spanner segments in systems in the old way (by overriding System::scanElements)? Or should I change the tree model to make the system contain the spanner segments?

Either is acceptable for now. We will want to look at this in more detail later because it will have consequences for how you write files to MSCX, and for my grand plan to store more than one layout in a file.

@MarcSabatella, to be clear, the `parent()` of the segments is still `System` (I think). We are talking about the new function `treeParent()`, which in most cases returns `parent()`, but not always.

Reply:
Made the system scan the spanner segments for now.